[move] Initial doc migration for 2024 (#16597)
MystenLabs/sui: external-crates/move/documentation/book/src/tuples.md

# Tuples and Unit

Move does not fully support tuples as one might expect coming from another language with them as a
[first-class value](https://en.wikipedia.org/wiki/First-class_citizen). However, in order to support
multiple return values, Move has tuple-like expressions. These expressions do not result in a
concrete value at runtime (there are no tuples in the bytecode), and as a result they are very

> limited: they can only appear in expressions (usually in the return position for a function); they

Reviewer: ```suggestion
limited: 
- They can only appear in expressions (usually in the return position for a function).
- They cannot be bound to local variables.
```

> cannot be bound to local variables; they cannot be stored in structs; and tuple types cannot be used

Reviewer: ```suggestion
- They cannot be bound to local variables.
- They cannot be stored in structs.
- Tuple types cannot be used to instantiate generics.
```

> to instantiate generics.

Reviewer: ```suggestion

```

    fun examples_with_function_calls() {
        let () = returns_unit();
        let (mut x, mut y): (bool, bool) = returns_2_values();
        let (mut a, mut b, mut c, mut d) = returns_4_values(&0);

        () = returns_unit();
        (x, y) = returns_2_values();
        (a, b, c, d) = returns_4_values(&1);
    }
}
```

For more details, see [Move Variables](./variables.md).

## Subtyping

Along with references, tuples are the only types that have

> [subtyping](https://en.wikipedia.org/wiki/Subtyping) in Move. Tuples do have subtyping only in the

Reviewer: ```suggestion
[subtyping](https://en.wikipedia.org/wiki/Subtyping) in Move. Tuples have subtyping only in the
```
